DESCRIPTION

```diff
@@ -37,7 +37,9 @@ Imports:
     tidyselect (>= 0.2.4),
     blob (>= 1.2.0),
     utils,
-    withr
+    withr,
+    tidyr,
```

Comment:
I think these probably should just be in suggests, and do conditional method registration.

Reply:
I don't understand how to make this work properly without loading tidyr... If `tidyr` isn't loaded then it doesn't find the function `expand()` and `complete()`. And loading `tidyr` into the namespace doesn't seem like an option.

Reply:
So the key idea is that we no longer use the NAMESPACE, and instead dynamically register using code in `.onLoad`. You can see the existing code in `.onLoad()` (although that's there for a different reason due to the problems with `dplyr::filter` being a generic and `stats::filter` not). You can get a few more details in `?vctrs::s3_register`, and otherwise let me know if you need more explanation.

Reply:
That's what I did in this commit https://github.com/tidyverse/dbplyr/pull/539/commits/5befb0c613dcb00a0bd78f85f956e9a2ec1f926e. The issue is that the user needs to still load `tidyr`. As I understand only if `expand()` is loaded the S3 registration actually works as otherwise R doesn't find the generic `expand()` itself from which it should dispatch.

Reply:
Right, I think that's what we want?